### PR TITLE
[E2E] Create/Recover Wallet, Send ZIL, Send ZRC-2, Send ETH, Send ERC-20, Network switch

### PR DESCRIPTION
**Pull request by NovacAndrei**
#775 [E2E] Test Create wallet flow
#776 [E2E] Test Recover wallet flow

#777 [E2E] Test Send ZIL native token flow
#778 [E2E] Test Send ETH native token flow 

#779 [E2E] Test Send ZRC-2 token flow
#780 [E2E] Test Send ERC-20 token flow

#782 [E2E] Test Network switch

### CODE REVIEW

File: e2e/tests/send/common.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    // Enter amount
+    await utils.elementTypeText('enter-amount', amount);
+    await utils.elementByIdTap('next');
+
+    // Confirm Transaction
+    await utils.elementByIdTap('confirm');
+
+    // Password Pin Screen
+    await utils.expectElementVisible('password-pin-screen');
+    await customKeyboard.typeWord('000000'); // enter pin code
```
Comment (georgeburduhos) [inline e2e/tests/send/common.ts:83]: make a const from pincode

File: e2e/tests/general/general.spec.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { generateWallet } from '../send/common';
+import { networkSwitch } from './network-switch';
+
+describe('General', () => {
+    /**
+     * Generate wallet
+     */
+    it('Generate wallet', async () => {
+        await generateWallet();
```
Comment (krisboit) [inline e2e/tests/general/general.spec.ts:9]: this should be like this:
```ts
// general.spec.ts
import { generateWalletTest } from '../send/common';
import { networkSwitchTest } from './network-switch';
describe('General', () => {
  generateWalletTest();
  networkSwitchTest();
}
```
```ts
// send/common.ts
export const generateWalletTest = () => {
  describe('Generate wallet ', () => {
    it('...', async () => {});
  }
}
```

File: e2e/tests/general/network-switch.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import * as utils from '../../utils/utils';
+
+/**
+ * Network Switch
+ */
+export const networkSwitch = async () => {
```
Comment (krisboit) [inline e2e/tests/general/network-switch.ts:6]: it's really hard to follow this...
The tests should be like this:
```ts
describe('Network switch', () => {
  ...
  test('it works ok on ethereum', () => {
    // call util function to do something reusable
    // check everything is fine

    // call util function to do something reusable
    // check everything is fine
  }

  test('it works on zilliqa', () => {
    // call util function to do something reusable
    // check everything is fine
  }
}
```

Try to use this pattern, it would be more ease to follow in case of a failing test

File: e2e/tests/general/general.spec.ts (hunk 2 above)
Comment (krisboit) [inline e2e/tests/general/general.spec.ts:9]: if anything is needed before test execution you should do it in:
```
beforeAll(() => {...});
```
or if it's needed before each test:
```
beforeEach(() => {...})
```

File: e2e/tests/general/network-switch.ts (hunk 3 above)
Comment (krisboit) [reply e2e/tests/general/network-switch.ts:6]: You have to separate the test logic and util function that you use in multiple tests.

File: e2e/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import { getProps } from 'detox-getprops';
+import * as customKeyboard from './custom-keyboard';
+
+/**
+ * Tap element by id
+ * @param key
+ */
+export const elementByIdTap = async (key: string) => element(by.id(key)).tap();
```
Comment (krisboit) [inline e2e/utils/utils.ts:8]: should be `tapElementById`

File: e2e/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * Tap element by label
+ * @param key
+ */
+export const elementByLabelTap = async (key: string) => element(by.label(key)).tap();
```
Comment (krisboit) [inline e2e/utils/utils.ts:14]: should be `tapElementByLabel`

File: e2e/utils/utils.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+ * @param label
+ */
+export const expectElementByIdToHaveLabel = async (key: string, label: string) =>
+    (expect(element(by.label(key))) as any).toHaveLabel(label);
+
+/**
+ * Get text of a component <Text />
+ * @param key
+ */
+export const getText = async (key: string) => ((await getProps(element(by.id(key)))) as any).text;
```
Comment (krisboit) [inline e2e/utils/utils.ts:58]: should be `getElementTextById`

File: e2e/utils/utils.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * Go back to the previous screen by pressing the left-arrow button
+ */
+export const goBack = async () => element(by.id('go-back')).tap();
```
Comment (krisboit) [inline e2e/utils/utils.ts:63]: this should be `tapBackButton`

File: e2e/utils/utils.ts [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * Go back to the previous screen
+ * @param currentScreen
+ * - Android: by using the device press back
+ * - iOS: swipe right on the current screen
+ *
+ * Experimental, need to be tested better
+ */
+export const pressBack = async (currentScreen: string) => {
```
Comment (krisboit) [inline e2e/utils/utils.ts:73]: this should be `deviceGoBack`

File: e2e/utils/utils.ts [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * Reload React Native App, Enter Pin Code, Dashboard is displayed
+ * @param pin the passcode of Moonlet
+ */
+export const realoadRNAndEnterPin = async (pin: string) => {
+    await device.reloadReactNative();
+
+    // Password Pin Screen
+    await expectElementVisible('password-pin-screen');
```
Comment (krisboit) [inline e2e/utils/utils.ts:101]: since this is a util function i don't know if we should do expects here ...
